src/app/input-container/input-container.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, OnDestroy } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { NumberApiServiceService } from '../numbers-api-service/number-api-service.service';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { MatDialog } from '@angular/material/dialog';
+import { FactDialogComponent } from '../fact-dialog/fact-dialog.component';
+import { DialogData, ApiResponse } from '../app.model';
+import { HttpErrorResponse } from '@angular/common/http';
+
+@Component({
+  selector: 'app-input-container',
+  templateUrl: './input-container.component.html',
+  styleUrls: ['./input-container.component.css']
+})
+export class InputContainerComponent implements OnDestroy {
+  public formProcessing = false;
+  public form = new FormGroup({
+    number: new FormControl(null, [Validators.required, Validators.pattern('^[0-9]*$')])
+  });
+
+  private readonly onDestroy$ = new Subject<void>();
+
+  constructor(
+    private service: NumberApiServiceService,
+    private dialog: MatDialog
+    ) { }
+
+  public findFact() {
+    if (this.form.controls.number.value && !this.formProcessing && this.form.valid) {
+      this.formProcessing = true;
+      this.service.getFact(this.form.controls.number.value)
+        .pipe(takeUntil(this.onDestroy$))
```

Comment:
There is no need to unsubscribe from observable, which is returned by `HttpClient`. Also, in this implementation it doesn't help a lot, as any invocation of `findFact` will create a new observable, which will be unsubscribed only after component is destroyed.
Could you remove unneeded code, please?

Reply:
Oh see. Did not know that Angular completes request after it was successful. Thanks for pointing this out will fix.